content/docs/visualization/template_reference.md

```diff
@@ -0,0 +1,107 @@
+---
+title: Template reference
+sort_rank: 6
+---
+
+# Template reference
+
+Prometheus has support for templating in both the summary/description field of
+alerts and also exposed as html endpoints. Templates have the ability to run
+queries against the databases, iterate over data, use conditionals, format data
+etc. This is based on the [go templating](http://golang.org/pkg/text/template/)
+system.
+
+## Data Structures
+
+The primary structure is the sample, defined as:
```

Comment:
Hmm, 'slice' is kinda Go specific (Python also has a notion of slice) and not a term readers may be familiar with.

This is also really docs for the `query` function, rather than the general data structures the various bits of templating system uses.

I think this should stay as is.